src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripContainerDesigner.cs

```diff
@@ -0,0 +1,502 @@
+
+        _selectionService ??= (ISelectionService)GetService(typeof(ISelectionService));
+
+        if (_topToolStripPanel is not null)
+        {
+            ToolStripPanelDesigner? panelDesigner = _designerHost?.GetDesigner(_topToolStripPanel) as ToolStripPanelDesigner;
+            panelDesigner?.ExpandTopPanel();
+        }
+
+        // Set ShadowProperties
+        TopToolStripPanelVisible = _toolStripContainer.TopToolStripPanelVisible;
+        LeftToolStripPanelVisible = _toolStripContainer.LeftToolStripPanelVisible;
+        RightToolStripPanelVisible = _toolStripContainer.RightToolStripPanelVisible;
+        BottomToolStripPanelVisible = _toolStripContainer.BottomToolStripPanelVisible;
+    }
+
+    public override void InitializeNewComponent(IDictionary defaultValues)
+    {
+        base.InitializeNewComponent(defaultValues);
+    }
+
+    protected override void OnPaintAdornments(PaintEventArgs pe)
+    {
+        try
+        {
+            _disableDrawGrid = true;
+
+            // we don't want to do this for the tab control designer
+            // because you can't drag anything onto it anyway.
+            // so we will always return false for draw grid.
+            base.OnPaintAdornments(pe);
+        }
+        finally
+        {
+            _disableDrawGrid = false;
+        }
+    }
+
+    /// <summary>
+    ///  Allows a designer to filter the set of properties
+    ///  the component it is designing will expose through the
+    ///  TypeDescriptor object.  This method is called
+    ///  immediately before its corresponding "Post" method.
+    ///  If you are overriding this method you should call
+    ///  the base implementation before you perform your own
+    ///  filtering.
+    /// </summary>
+    protected override void PreFilterProperties(IDictionary properties)
+    {
+        PropertyDescriptor? propertyDescriptor;
+
+        base.PreFilterProperties(properties);
+
+        // Handle shadowed properties
+        string[] shadowProps =
+        [
+            "TopToolStripPanelVisible",
+            "LeftToolStripPanelVisible",
+            "RightToolStripPanelVisible",
+            "BottomToolStripPanelVisible"
+        ];
+
+        Attribute[] empty = [];
+
+        for (int i = 0; i < shadowProps.Length; i++)
+        {
+            propertyDescriptor = (PropertyDescriptor?)properties[shadowProps[i]];
+            if (propertyDescriptor is not null)
+            {
+                properties[shadowProps[i]] = TypeDescriptor.CreateProperty(typeof(ToolStripContainerDesigner), propertyDescriptor, empty);
+            }
+        }
+    }
+
+    private void AddPanelSelectionGlyph(ToolStripPanelDesigner designer, SelectionManager selectionManager)
+    {
+        // Now create SelectionGlyph for the panel and add it
+        if (designer is not null)
+        {
+            Glyph? childGlyph = designer.GetGlyph();
+            if (childGlyph is not null && _selectionService is not null)
```

Comment:
Please invert and combine the if statements:
```suggestion
        if (designer is null || designer.GetGlyph() is not { } childGlyph || _selectionService is null)
        {
            return;
        }
```